### ISSUE TITLE
kitchen converge throws an error when cookbook generator is configured

### ISSUE THREAD

**Issue by teknofire** [chef/chef-workstation#688]
## Description
Running `kitchen converge` with `chefcli.generator_cookbook` in `~/.chef/knife.rb` throws the following error:

```
NoMethodError: undefined method `generator_cookbook' for nil:NilClass
/home/USERNAME/.chef/knife.rb:11:in `from_string'
Relevant file content:
11: chefcli.generator_cookbook "/home/USERNAME/cookbooks/my_generator"
```

Other `chef` commands such as `chef generate cookbook test_cookbook` will complete successfully. 

We were able to work around the problem by changing `knife.rb` to use `respond_to?`:
```
if respond_to?(:chefcli)
  chefcli.generator_cookbook "/path/to/my_generator"
end
```

## To Reproduce
1. Generate a new cookbook generator using `chef generate generator my_generator`
2. Follow the instructions to add `chefcli.generator_cookbook "/path/to/my_generator"` to `knife.rb`
3. Go to any other cookbook on the system and run `kitchen converge`, it doesn't have to be one that was created using the cookbook generator.

## Chef Workstation Version
0.9.42

## Platform Version
OSX, Ubuntu 18.04

**Comment by tyler-ball** [comment]
Thanks for the report and repro steps @teknofire!

**Comment by antima-gupta** [comment]
This issue was occurring because of `config_context for chefcli` was not added in `Chef Workstation Version: 0.9.42`. This issue has fixed in the PR[#8911](https://github.com/chef/chef/pull/8911) .Please download the latest Chef Workstation.

Here is the details :

## Recipe:
```
sudo '20_test' do
  groups 'testuser1'
  commands ['/bin/ls']
  nopasswd true
end
```
## Command: 
   kitchen converge

## Log:

-----> Starting Test Kitchen (v2.5.2)
-----> Creating <default-ubuntu-1804>...
       Bringing machine 'default' up with 'virtualbox' provider...
       ==> default: Importing base box 'bento/ubuntu-18.04'...
==> default: Matching MAC address for NAT networking...
       ==> default: Checking if box 'bento/ubuntu-18.04' version '202005.21.0' is up to date...
       ==> default: Setting the name of the VM: kitchen-test_cookbook-default-ubuntu-1804-4a48c848-1bb3-4546-ba11-5fc7696fc486
       ==> default: Clearing any previously set network interfaces...
       ==> default: Preparing network interfaces based on configuration...
           default: Adapter 1: nat
       ==> default: Forwarding ports...
           default: 22 (guest) => 2222 (host) (adapter 1)
       ==> default: Running 'pre-boot' VM customizations...
       ==> default: Booting VM...
       ==> default: Waiting for machine to boot. This may take a few minutes...
           default: SSH address: 127.0.0.1:2222
           default: SSH username: vagrant
           default: SSH auth method: private key
           default: Warning: Remote connection disconnect. Retrying...
           default: Warning: Remote connection disconnect. Retrying...
           default: Warning: Remote connection disconnect. Retrying...
           default: Warning: Remote connection disconnect. Retrying...
           default: Warning: Remote connection disconnect. Retrying...
           default: 
           default: Vagrant insecure key detected. Vagrant will automatically replace
           default: this with a newly generated keypair for better security.
           default: 
           default: Inserting generated public key within guest...
           default: Removing insecure key from the guest if it's present...
           default: Key inserted! Disconnecting and reconnecting using new SSH key...
       ==> default: Machine booted and ready!
       ==> default: Checking for guest additions in VM...
       ==> default: Setting hostname...
       ==> default: Mounting shared folders...
           default: /tmp/omnibus/cache => /home/antima/.kitchen/cache
       ==> default: Machine not provisioned because `--no-provision` is specified.
       [SSH] Established
       Vagrant instance <default-ubuntu-1804> created.
       Finished creating <default-ubuntu-1804> (10m54.72s).
-----> Converging <default-ubuntu-1804>...
       Preparing files for transfer
       Installing cookbooks for Policyfile /home/antima/workspace/learn-chef/cookbooks/test_cookbook/Policyfile.rb using `chef install`
       Installing cookbooks from lock
       Installing test_cookbook 0.1.0
       Preparing dna.json
       Exporting cookbook dependencies from Policyfile /tmp/default-ubuntu-1804-sandbox-20200619-3523-ynj93h...
       Exported policy 'test_cookbook' to /tmp/default-ubuntu-1804-sandbox-20200619-3523-ynj93h
       
       To converge this system with the exported policy, run:
         cd /tmp/default-ubuntu-1804-sandbox-20200619-3523-ynj93h
         chef-client -z
       Removing non-cookbook files before transfer
       Preparing validation.pem
       Preparing client.rb
-----> Installing Chef install only if missing package
       Downloading https://omnitruck.chef.io/install.sh to file /tmp/install.sh
       Trying wget...
       Download complete.
       ubuntu 18.04 x86_64
       Getting information for chef stable  for ubuntu...
       downloading https://omnitruck.chef.io/stable/chef/metadata?v=&p=ubuntu&pv=18.04&m=x86_64
         to file /tmp/install.sh.1939/metadata.txt
       trying wget...
       sha1 6c9268d682eb913e3c050afdc4bac793454ba106
       sha256 a6c7808485b9395db1e50cce45d9bec733689cfcfec0c56bd04f60be5771d540
       url  https://packages.chef.io/files/stable/chef/16.2.44/ubuntu/18.04/chef_16.2.44-1_amd64.deb
       version  16.2.44
       downloaded metadata file looks valid...
       /tmp/omnibus/cache/chef_16.2.44-1_amd64.deb exists
       Comparing checksum with sha256sum...
       
       WARNING WARNING WARNING WARNING WARNING WARNING WARNING WARNING WARNING
       
       You are installing a package without a version pin.  If you are installing
       on production servers via an automated process this is DANGEROUS and you will
       be upgraded without warning on new releases, even to new major releases.
       Letting the version float is only appropriate in desktop, test, development or
       CI/CD environments.
       
       WARNING WARNING WARNING WARNING WARNING WARNING WARNING WARNING WARNING
       
       Installing chef 
       installing with dpkg...
       Selecting previously unselected package chef.
(Reading database ... 42697 files and directories currently installed.)
       Preparing to unpack .../cache/chef_16.2.44-1_amd64.deb ...
       Unpacking chef (16.2.44-1) ...
       Setting up chef (16.2.44-1) ...
       Thank you for installing Chef Infra Client! For help getting started visit https://learn.chef.io
       Transferring files to <default-ubuntu-1804>
       +---------------------------------------------+
       ✔ 2 product licenses accepted.
       +---------------------------------------------+
       Starting Chef Infra Client, version 16.2.44
       Creating a new client identity for default-ubuntu-1804 using the validator key.
       Using policy 'test_cookbook' at revision '06a24084d0feee36b5361974ec61a89fd2b556af4808862c628b090a7d750be0'
       resolving cookbooks for run list: ["test_cookbook::default@0.1.0 (640a5bd)"]
       Synchronizing Cookbooks:
         - test_cookbook (0.1.0)
       Installing Cookbook Gems:
       Compiling Cookbooks...
       Converging 1 resources
       Recipe: test_cookbook::default
         * sudo[20_test] action create
           * directory[/etc/sudoers.d/] action create (up to date)
           * template[/etc/sudoers.d/20_test] action create
             - create new file /etc/sudoers.d/20_test
             - update content in file /etc/sudoers.d/20_test from none to 45752f
             --- /etc/sudoers.d/20_test 2020-06-19 09:55:59.001896000 +0000
             +++ /etc/sudoers.d/.chef-20_test20200619-2037-14jv38r  2020-06-19 09:55:58.909849999 +0000
             @@ -1 +1,5 @@
             +# This file is managed by Chef Infra Client. Changes will be overwritten.
             +
             +%testuser1 ALL=(ALL) NOPASSWD:/bin/ls
             +%testuser1 ALL=(ALL) NOPASSWD:/bin/pwd
             - change mode from '' to '0440'
         
       
       Running handlers:
       Running handlers complete
       Chef Infra Client finished, 2/3 resources updated in 17 seconds
       Downloading files from <default-ubuntu-1804>
       Finished converging <default-ubuntu-1804> (4m34.74s).
-----> Test Kitchen is finished. (16m45.49s)

**Comment by dheerajd-msys** [comment]
This issue has been fixed via https://github.com/chef/chef/pull/8911 and released in `chef-client v15.4.4+` shipped in `chef-workstation v0.10.37+`
